src/components/PopupWithForm.js

```diff
@@ -22,9 +23,31 @@ export default class PopupWithForm extends Popup {
     this._modalForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       const inputValues = this._getInputValues();
-      console.log("Input values before submit:", inputValues);
+      this.renderSaving(true);
       this._handleFormSubmit(inputValues);
+      this.renderSaving(false);
     });
     super.setEventListeners();
   }
+
+  stopSaving(popup) {
+    popup._modalForm.querySelector(".modal__save-button").textContent =
+      popup._buttonText;
+    console.log("Found _modalForm and done");
+    popup.close();
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`

Reply:
I'm confused about this. Would you mind specify which part of the code in index.js this would help with? Thank you.

Reply:
> I'm confused about this. Would you mind specify which part of the code in index.js this would help with? Thank you.

The profile opening:

```js

  const data = newUserInfo.getUserInfo();
  modalName.value = data.name;
  modalJob.value = data.job;
```

Can be replaced with this:


```js
  const data = newUserInfo.getUserInfo();
  profileModal.setInputValues(data)  // you need to change the `html` attributes `name` to achieve that 

```

Reply:
Thank you!